services/QuillLMS/app/controllers/api/v1/activity_sessions_controller.rb

```diff
@@ -85,7 +85,9 @@ def destroy
     concept_results_to_save = @concept_results.map { |c| concept_results_hash(c) }.reject(&:empty?)
     return if concept_results_to_save.empty?
```

Comment:
You can remove the `return` above since `each` does that check. You could also consider removing the interim variable and just chaining it all (take or leave this)

```ruby
@concept_results
  .map {|cr| concept_results_hash(cr) }
  .reject(&:empty?)
  .each {|crh| SaveActivitySessionConceptResultsWorker.perform_async(crh) }

Reply:
Making it a single chain seems reasonable.  I'll make that update.